### PR TITLE
Move l3box functions to stable

### PR DESCRIPTION
**Pull request by josephwright**
As discussed at the team meeting, including the rename (old names unused in TL)

### CODE REVIEW

File: l3kernel/l3box.dtx [hunk 1]
```diff
@@ -874,6 +874,68 @@
...
+%   unless \cs{box_clip:N} is subsequently applied.
+%   The updated \meta{box} is an
+%   hbox, irrespective of the nature of the \meta{box} before the trim
+%   operation is applied.
+%   The behavior of the operation where the trims requested is
+%   greater than the size of the box is undefined.
+% \end{function}
+%
+% \begin{function}[added = 2019-01-23]
+%   {
```
Comment (FrankMittelbach) [inline l3kernel/l3box.dtx:921]: shouldn't that give the current day? well, I guess there are arguments for both so not sure

File: l3kernel/l3box.dtx [hunk 2]
```diff
@@ -874,6 +874,68 @@
...
+%     \cs{box_set_viewport:Nnnnn} \meta{box} \Arg{llx} \Arg{lly} \Arg{urx} \Arg{ury}
+%   \end{syntax}
+%   Adjusts the bounding box of the \meta{box} such that it has lower-left
+%   co-ordinates (\meta{llx}, \meta{lly}) and upper-right co-ordinates
+%   (\meta{urx}, \meta{ury}). All four co-ordinate positions are
+%   \meta{dim exprs}. Material outside of the bounding box is
+%   still displayed in the output unless \cs{box_clip:N} is
+%   subsequently applied.
+%   The updated \meta{box} is an
+%   hbox, irrespective of the nature of the \meta{box} before the viewport
```
Comment (FrankMittelbach) [inline l3kernel/l3box.dtx:935]: is the nature of the box changed or (more likely I guess) is another boxing level added? If so mit might be worth pointing out that you generate new boxing levels each time you apply such an operation.

**Review by FrankMittelbach — APPROVED** [review]
minor comment/query

File: l3kernel/l3box.dtx (hunk 1 above)
Comment (josephwright) [reply l3kernel/l3box.dtx:921]: The code in the current form was added then to `l3candidates`: the move to stable doesn't affect that. So I thought this made the most sense: will be different for a move of `l3graphics`, for example, as that's not currently in `l3kernel`.

Comment (FrankMittelbach) [reply l3kernel/l3box.dtx:921]: that's why I said there are arguments both ways

File: l3kernel/CHANGELOG.md [hunk 3]
```diff
@@ -7,9 +7,19 @@ this project uses date-based 'snapshot' version identifiers.
...
+### Changed
+- `\box_set_trim:Nnnnn` and `\box_set_viewport:Nnnnn` moved to
+  stable kernel
+
 ### Fixed
 - Incorrect internal name in `\pdf_pagesize_gset:nn`
 
+### Removed
+- Experimental `\box_clip:N` and variants
+
```
Comment (muzimuzhi) [inline l3kernel/CHANGELOG.md:22]: To better reflect it's a renaming, I personally perfer
```suggestion
### Changed
- Experimental `\box_clip:N` and variants renamed to `\box_set_clipped:N` and
  variants
- `\box_set_clipped:N`, `\box_set_trim:Nnnnn`, and `\box_set_viewport:Nnnnn`
  moved to stable kernel

### Fixed
- Incorrect internal name in `\pdf_pagesize_gset:nn`

```

Comment (josephwright) [reply l3kernel/CHANGELOG.md:22]: I see what you mean, but the idea in  a formal change log is that *anything* that 'appears' goes down in the `Added` and *anything* that 'disappears' goes down in `Removed`: the fact it's a rename is a detail from that point of view.

Comment (josephwright) [reply l3kernel/CHANGELOG.md]: I've clarified the renaming in the changelog

**Pull request merged**